docs/concepts/ksql-architecture.rst

```diff
@@ -0,0 +1,368 @@
+   :alt: Diagram showing KSQL embedded in an application
+   :align: center
+
+In embedded mode you can:
+
+* Embed your KSQL statements and queries directly in your Java application
+* Generate and execute KSQL queries through the Java API
+* Version-control your queries and transformations as code
+
+In embedded mode, you deploy a JVM-based application by using the application
+framework of your choice, like Spring, Grails, Jersey, VertX, or Ratpack. Your 
+application executes KSQL queries without starting a separate KSQL cluster.
+You embed the KSQL engine in your application, and you scale the application and
+its stream processing the same way that you scale a Kafka Streams application or
+KSQL cluster, by using a consumer group.
+
+Dedicating Resources
+====================
+
+Join KSQL engines to the same *service pool* by using the ``ksql.service.id``
+property. The following diagram shows a Kafka cluster with separate workloads
+for a ``finance`` pool and a ``sales`` pool. For more information, see
+:ref:`ksql-service-id`.
+
+.. image:: ../img/ksql-dedicating-resources.png
+   :alt: Diagram showing how to join KSQL engines to the same service pool
+   :align: center
+
+To scale out, just add more KSQL server instances. There's no master node or 
+coordination among them required. For more information, see
+:ref:`ksql_capacity_planning`.
+
+KSQL Query Lifecycle
+********************
+
+To create a streaming application with KSQL, you write KSQL statements and
+queries. Each statement and query has a lifecycle with the following steps:
+
+#. You register a KSQL stream or table from an existing Kafka topic with a DDL
+   statement, like CREATE STREAM <my-stream> WITH <topic-name>.
+#. You express your app by using a DML statement, like CREATE TABLE AS SELECT
+   FROM <my-stream>.
+#. KSQL parses your statement into an abstract syntax tree (AST).
+#. KSQL uses the AST to create the logical plan for your statement.
+#. KSQL uses the logical plan to create the physical plan for your statement.
+#. KSQL generates and runs the Kafka Streams application.
+#. You manage the application as a STREAM or TABLE with its corresponding
+   persistent query.
+
+.. image:: ../img/ksql-query-lifecycle.gif
+   :alt: Diagram showing how the KSQL query lifecycle for a KSQL statement
+   :align: center
+
+Register the Stream
+===================
+
+Register a stream or table by using the DDL statements, CREATE STREAM and
+CREATE TABLE. For example, the following KSQL statement creates a stream named
+``authorization_attempts`` that's backed by a topic named ``authorizations``.
+
+.. code:: sql
+
+    CREATE STREAM authorization_attempts 
+      (card_number VARCHAR, attemptTime BIGINT, ...)
+      WITH (kafka_topic='authorizations', value_format=‘JSON’);
+
+KSQL writes DDL and DML statements to the *command topic*. Each KSQL
+Server reads the statement from the command topic, parsing and analyzing
+it.
+
+.. image:: ../img/ksql-deploy-command-topic.gif

```

Comment:
this animated graphic is a bit tough to follow. i feel like i'm watching a video on mute - not sure what's going on - or what step i'm on. 

![image](https://user-images.githubusercontent.com/11722533/50112856-2cb22b00-01f5-11e9-9195-ba0a85d9ec14.png)